pontoon/base/static/js/pontoon.js

```diff
@@ -531,16 +531,21 @@
               startEditing();
             }
             break;
+          case "DISCARD":
+            /**
+             * Messsage is sent when the locale maintainer discards translation of the entity.
+             */
 
+            var entity = message.value.id ? Pontoon.entities[message.value.id] : $('.pontoon-editable-toolbar')[0].target.entity,
+                translationValue = message.value.translation,
+                translation = translationValue !== undefined ? translationValue : message.value;
```

Comment:
This block is never called... And also doesn't do anything except for burning some memory and time. ;)